### DISCUSSION TITLE
Consider upgrading asciinema player in docs

### DISCUSSION THREAD

**Discussion by ku1ik** [owncast/owncast#3294] [General Owncast discussion]
Hey there,

asciinema creator here. While browsing through docs I saw you use asciinema player for some of the command line demos. That's awesome.

I noticed you use older version of the player (v2.x), and thought you may benefit from upgrading to v3, which is significantly improved. One (of _many_) improvements is the player now adjusts font size to the available size in the container element which results in better utilization of the space on the page, and it's also responsive, i.e. deals with browser window resize or screen orientation changes without problems (contrary to old v2).

I thought I'll let you know, and if you decide to upgrade feel free to ping me, happy to assist.

Stream on!

**Comment by gabek** [comment]
Hi there, @ku1ik, thanks for checking in!

So I've got some not positive news. I used to try to keep our asciinema content up to date to reflect updates to Owncast (but clearly not keeping the asciinema player updated), but I stopped doing it at one point because it was additional manual effort. Recently I wanted to fix this problem and automate the entire thing, so I made the change to use VHS instead, since you can [automate the generation](https://github.com/owncast/owncast.github.io/blob/v0.1.2/scripts/preview-generator/installer.tape) and spit it out as a GIF. I was just sick of the example content showing ancient content, and I was sick of manually creating the content :)

So with the next release, it's being swapped out for that VHS-generated GIF. It's certainly not as feature packed as asciinema, but I'm not sure how many people took advantage of that outside of just pressing play.

There's been[ discussion to use VHS to generate asciinema-compatible output](https://github.com/charmbracelet/vhs/issues?q=is%3Aissue+asciinema+), and if that ever went in, and I could continue to automate it via the tape file, I might very well use the asciinama player again.

**Comment by ku1ik** [comment]
Fair enough, thanks!

**Reply by ku1ik** [reply]
Btw, there's also https://github.com/k9withabone/autocast which generates asciicast files from a script, so you can easily regenerate a recording without any manual work.
